skfp/model_selection/splitters/pubchem_split.py

```diff
@@ -0,0 +1,456 @@
+    n_jobs: int = 5,
+) -> Union[
+    tuple[
+        Sequence[Union[str, Mol]],
+        Sequence[Union[str, Mol]],
+        Sequence[Union[str, Mol]],
+        Sequence[Sequence[Any]],
+    ],
+    tuple[Sequence, ...],
+    tuple[Sequence[int], Sequence[int], Sequence[int]],
+]:
+    """
+    Split using PubChem literature data.
+
+    This is a time (chronological) split, using first literature date in PubChem
+    available for each molecule. Molecules are partitioned deterministically, with
+    the newest ones assigned to the test subset and the rest to the training subset.
+
+    In case of no data available (no literature or no PubChem entry) the behavior
+    is governed by the ``not_found_behavior`` parameter. Full DataFrame with columns
+    ``["SMILES", "split", "year"]`` can also be returned.
+
+    Note that this split can be very slow for large number of molecules, since requests
+    to PubChem frequently fail due to the "servers overloaded" error.
+
+    The split fractions (train_size, valid_size, test_size) must sum to 1.
+
+    Parameters
+    ----------
+    data : sequence
+        A sequence representing either SMILES strings or RDKit `Mol` objects.
+
+    additional_data: sequence
+        Additional sequences to be split alongside the main data, e.g. labels.
+
+    train_size : float, default=None
+        The fraction of data to be used for the train subset. If None, it is set
+        to 1 - test_size - valid_size. If valid_size is not provided, train_size
+        is set to 1 - test_size. If train_size, test_size and valid_size aren't
+        set, train_size is set to 0.8.
+
+    valid_size : float, default=None
+        The fraction of data to be used for the test subset. If None, it is set
+        to 1 - train_size - valid_size. If train_size, test_size and valid_size
+        aren't set, train_size is set to 0.1.
+
+    test_size : float, default=None
+        The fraction of data to be used for the validation subset. If None, it is
+        set to 1 - train_size - valid_size. If valid_size is not provided, test_size
+        is set to 1 - train_size. If train_size, test_size and valid_size aren't set,
+        test_size is set to 0.1.
+
+    not_found_behavior : {"train", "test", "remove"}, default="test"
+        What to do with molecules not found in PubChem, or if molecule does not have
+        any literature data available. Such molecules can be put into train set,
+        test set, or removed altogether.
+
+    return_indices : bool, default=False
+        Whether the method should return the input object subsets, i.e. SMILES strings
+        or RDKit ``Mol`` objects, or only the indices of the subsets instead of the data.
+
+    return_indices : bool, default=False
+        Whether the method should return the input object subsets, i.e. SMILES strings
+        or RDKit `Mol` objects, or only the indices of the subsets instead of the data.
+
+    n_jobs : int, default=5
```

Comment:
You should check this somewhere and explicitly limit this, so that we don't exceed 5 requests per second

Reply:
pubchem has restriction 5 requests per second  (https://pubchem.ncbi.nlm.nih.gov/docs/pug-rest)